pages/doc.js

```diff
@@ -55,34 +57,15 @@ export default class Documentation extends Component {
   }
 
   loadStateFromURL = () => {
-    const { pathname } = window.location
-    const sectionURL = pathname.split('/')[2] // match section from URL
-    const sectionIndex = sidebar.findIndex(
-      section => (section.slug || kebabCase(section.name)) === sectionURL
-    )
-    if (sectionIndex === -1) {
-      sectionURL
-        ? this.setState({ pageNotFound: true })
-        : this.onSectionSelect(0)
-    } else {
-      const fileURL = pathname.split('/')[3] // match file from URL
-      const sectionFiles = flatten(sidebar[sectionIndex].files)
-      const fileIndex = sectionFiles.findIndex(
-        file => kebabCase(file.slice(0, -3)) === fileURL
-      )
-      if (fileIndex === -1) {
-        fileURL
-          ? this.setState({ pageNotFound: true })
-          : this.onSectionSelect(sectionIndex)
-      } else {
-        this.loadFile({
-          section: sectionIndex,
-          file: sectionFiles[fileIndex],
-          parseHeadings: true,
-          pageNotFound: false
-        })
-      }
-    }
+    let path = window.location.pathname.split(PATH_SEPARATOR)
+    let length = path.length
+    let { file, indexes } = SidebarMenuHelper.getFileFromUrl(path)
```

Comment:
can we make this function actually return what is needed for the loadFile below? why should I care here about some indexes I don't know anything about. It looks like I care about section, subsection, and file. Let's return them?

Reply:
loadFile works with files of different levels in sidebar.json, so this function needs all the information.